src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1304,8 +1303,15 @@ bool Arguments::add_property(const char* prop, PropertyWriteable writeable, Prop
 #endif
 
   if (strcmp(key, "java.compiler") == 0) {
-    process_java_compiler_argument(value);
-    // Record value in Arguments, but let it get passed to Java.
+    // we no longer support java.compiler system property, log a warning and let it get
+    // passed to Java, like any other system property
+    if (strlen(value) == 0 || strcasecmp(value, "NONE") == 0) {
+        // for applications using NONE or empty value, log a more informative message
+        warning("The java.compiler system property is obsolete and no longer supported,"
+                " use -Xint if you want to run the application in interpreted-only mode.");
+    } else {
+        warning("The java.compiler system property is obsolete and no longer supported.");
```

Comment:
This looks okay although the warning for the empty value and "NONE" case will probably wrap as it's very long. Today, -Djava.compiler=foo is ignored, no warning, seems good to emit a warning about that too.

Reply:
Hello Alan,

> This looks okay although the warning for the empty value and "NONE" case will probably wrap as it's very long.

Agreed, it's a bit long. I can't think of a simpler message though. If anyone has a suggestion, I'll update accordingly.

Reply:
Just drop the detailed explanation, they can check the help for -Xint.

"The java.compiler system property is obsolete and no longer supported, use -Xint".

Reply:
Thank you Roger, updated the PR with this suggested change.